hoomd/pytest/test_simulation.py

```diff
@@ -0,0 +1,196 @@
+                    assert getattr(getattr(gsd_snap, attr), prop) == \
+                        getattr(getattr(hoomd_snap, attr), prop)
+            else:
+                if hoomd_snap.exists:
+                    np.testing.assert_allclose(
+                        getattr(getattr(gsd_snap, attr), prop),
+                        getattr(getattr(hoomd_snap, attr), prop)
+                    )
+
+
+def assert_equivalent_boxes(box1, box2):
+    assert box1.Lx == box2.Lx
+    assert box1.Ly == box2.Ly
+    assert box1.Lz == box2.Lz
+    assert box1.xy == box2.xy
+    assert box1.xz == box2.xz
+    assert box1.yz == box2.yz
+
+
+def random_inds(n):
+    return np.random.choice(np.arange(n),
+                            size=int(n * np.random.rand()),
+                            replace=False)
+
+
+def test_initialization(device, simulation_factory, get_snapshot):
+    with pytest.raises(TypeError):
+        sim = hoomd.Simulation()
+
+    sim = hoomd.Simulation(device)
+
+
+def test_device_property(device):
+    sim = hoomd.Simulation(device)
+    assert sim.device is device
+
+    with pytest.raises(ValueError):
+        sim.device = device
+
+
+def test_allows_compute_pressure(device, get_snapshot):
+    sim = hoomd.Simulation(device)
+    assert sim.always_compute_pressure is False
+    with pytest.raises(RuntimeError):
+        sim.always_compute_pressure = True
+    sim.create_state_from_snapshot(get_snapshot())
+    sim.always_compute_pressure = True
+    assert sim.always_compute_pressure is True
+
+
+def test_run(simulation_factory, get_snapshot, device):
+    sim = hoomd.Simulation(device)
+    with pytest.raises(RuntimeError):
+        sim.run(1)  # Before setting state
+
+    sim = simulation_factory(get_snapshot())
+    sim.run(1)
+
+    assert sim.operations.scheduled
+
+
+def test_timestep(simulation_factory, get_snapshot, device):
+    sim = hoomd.Simulation(device)
+    assert sim.timestep is None
+
+    initial_steps = 10
+    sim.timestep = initial_steps
+    assert sim.timestep == initial_steps
+    sim.create_state_from_snapshot(get_snapshot())
+    assert sim.timestep == initial_steps
+
+    with pytest.raises(RuntimeError):
+        sim.timestep = 20
+
+
+def test_run_with_timestep(simulation_factory, get_snapshot, device):
+    sim = hoomd.Simulation(device)
+    sim.create_state_from_snapshot(get_snapshot())
+    sim.operations.schedule()
+
+    steps = 0
+    n_step_list = [1, 10, 100]
+    for n_steps in n_step_list:
+        steps += n_steps
+        sim.run(n_steps)
+        assert sim.timestep == steps
+
+    assert sim.timestep == sum(n_step_list)
+
+
+_state_args = [((10, ['A']), 10),
+               ((5, ['A', 'B']), 20),
+               ((50, ['A', 'B', 'C']), 4),
+               ((100, ['A', 'B']), 30)]
+
+
+@pytest.fixture(scope="function", params=_state_args)
+def state_args(request):
+    return deepcopy(request.param)
+
+
+@skip_gsd
+def test_state_from_gsd(simulation_factory, get_snapshot,
```

Comment:
This test fails in MPI: https://dev.azure.com/glotzerlab/hoomd-blue/_build/results?buildId=3558&view=logs&j=959d7a94-bee1-5f30-271f-627c279bcb55&t=e40ef08a-fd92-5fa7-27f4-e5706e667ef3&l=332